pyvrp/cpp/ProblemData.h

```diff
@@ -130,15 +149,25 @@ class ProblemData
      *     Depot associated with these vehicles.
      * fixed_cost
      *     Fixed cost of using a vehicle of this type.
+     * tw_early
+     *     Start of the vehicle type's shift, if specified.
+     * tw_late
+     *     End of the vehicle type's shift, if specified.
      */
     struct VehicleType
     {
         Load const capacity;        // This type's vehicle capacity
         size_t const numAvailable;  // Available vehicles of this type
         size_t const depot = 0;     // Departure and return depot location
-        Cost fixedCost = 0;         // Fixed cost of using this vehicle type
+        Cost const fixedCost;       // Fixed cost of using this vehicle type
+        std::optional<Duration> const twEarly;  // Start of shift
+        std::optional<Duration> const twLate;   // End of shift
 
-        VehicleType(Load capacity, size_t numAvailable, Cost fixedCost = 0);
+        VehicleType(Load capacity,
+                    size_t numAvailable,
+                    Cost fixedCost = 0,
+                    std::optional<Duration> twEarly = std::nullopt,
+                    std::optional<Duration> twLate = std::nullopt);
     };
```

Comment:
What I'm not sure about:

- Calling this a shift time window, and use `twEarly` and `twLate`. It makes sense to do for consistency, since we also call the depot open/close times `twEarly` and `twLate`. But `shiftStart` and `shiftEnd` could also work.
- Using `None`/`std::optional` for this. The issue is that we'd like to default to the depot's time window if this is not provided, but have no clean way to tell whether the data is provided or not if we use integers (e.g., `0` could be the default, or it could be given explicitly by the user). We could also say "anything but (0, 0) indicates user arguments, but (0, 0) means default to the depot", but I think that's not as explicit as this.

Reply:
- I see a shift essentially as a vehicle time window, so keeping `tw_early` and `tw_late` is nice for consistency. But I can also see the value of calling it shiftStart and shiftEnd, since that better captures the meaning. I don't really have a preference for this.
- `(0, 0)` is a valid (although practically useless) shift time window so I don't think we should use that as default. I think the `None`s here are fine!